# Assignment 2
# Student ID: 1221140 
# Name: Ibrahim Abuhashhash

#### Import necessary libarries

In [67]:
import pandas as pd 
import numpy as np

## Loading Datasets

### DataFrame 1

In [68]:
df1 = pd.read_csv("palestine_food_prices.csv")

### DataFrame 2 

In [69]:
df2 = pd.read_excel("palestine_political_events.xlsx")

### Unify Units for DataFrame 1

In [70]:
def price_per_kilo(row):
    if row['unit'] == 'Cubic meter':
        return row['price']/1000
    elif row['unit'] == 'G':
        return row['price']*1000
    elif row['unit'] == 'KG':
        return row['price']
    elif row['unit'] == 'L':
        return row['price']

    if len(row['unit'].split())==2:
        parts = row['unit'].split()
        quantity = float(parts[0])
        unit_type = parts[1]

        if unit_type == 'G':
            return row['price']*(1000 / quantity)
        elif unit_type == 'KG':
            return row['price']/quantity
        elif unit_type == 'L':
            return row['price']/quantity
        elif unit_type == 'Cubic meter':
            return row['price']/(1000 * quantity)

    return None

df1['price_per_kg'] = df1.apply(price_per_kilo, axis=1)
df1['price_per_kg'] = df1['price_per_kg'].round(3)            

## 1. When did the food price monitoring begin in Gaza and West Bank?

In [71]:
df1['date'] = pd.to_datetime(df1['date']) 
wb_date = df1[df1['region'] == 'West Bank']
gaza_date = df1[df1['region'] == 'Gaza Strip']

wb_first_date = wb_date['date'].min()
gaza_first_date = gaza_date['date'].min()

print(f"The food price monitoring in the West Bank first began on: {wb_first_date}")
print(f"The food price monitoring in Gaza Strip first began on: {gaza_first_date}")

The food price monitoring in the West Bank first began on: 2014-05-15 00:00:00
The food price monitoring in Gaza Strip first began on: 2007-01-15 00:00:00


The date of the food price monitoring in the West Bank first began on 2014-05-15, While the date of the food price monitoring in the Gaza Strip first began on 2007-01-15. which means that the food price monitoring in gaza began 7 years earlier from the West Bank's. 

## 2. Compare the average price of "meat,fish and eggs" category between Gaza and West Bank? 

In [72]:
avg_price = df1[df1['category'] == 'meat, fish and eggs'].groupby('region')['price_per_kg'].mean()
avg_price

region
Gaza Strip    25.094767
West Bank     30.825238
Name: price_per_kg, dtype: float64

The average price in the Gaza Strip for meat, fish and eggs cateogry is 25 ILS, while the average price for the same cateogory in the West Bank is 30 ILS. which means that the West Bank is more expensive than Gaza's by 5 ILS. 

## 3. Which commodities have experienced the highest price volatility (extreme changes in price) over time?

In [73]:
price_volatility = df1.groupby('commodity')['price_per_kg'].std()
high_volatility_commodities = price_volatility.sort_values(ascending=False)
high_volatility_commodities.head(10)

commodity
Meat (goat, with bones)    14.946909
Fuel (petrol-gasoline)     13.055572
Meat (beef)                12.053133
Fish (red snapper)          5.779109
Milk (powder)               5.370760
Tea                         5.155695
Cheese (goat)               4.386487
Oil (olive)                 3.795847
Yogurt                      3.097520
Eggplants (large)           2.998813
Name: price_per_kg, dtype: float64

The commodities that have experienced extreme changes in price over time are the following: Meat (goat, with bones), Fuel (petrol-gasoline), meat(goat, Meat (beef). those three commodities are more sensitive to change than others.

## 4. Compare the total Number of Fatalities between different years. is it increasing or decreasing? 

In [77]:
df2.groupby('Year')['Number of Fatalities'].sum()
fatalities_by_year = df2.groupby('Year')['Number of Fatalities'].sum()
print(fatalities_by_year)
cor = fatalities_by_year.corr(df2.groupby('Year')['Number of Events'].sum())
print(f"the correlation to see if it is inreasing or decreasing:{cor.round(4)}")

Year
2016      164
2017      108
2018      199
2019      146
2020       43
2021      312
2022      216
2023    22607
2024    24873
Name: Number of Fatalities, dtype: int64
the correlation to see if it is inreasing or decreasing:0.947


The data shows the total number of fatalities of each year from 2016 to 2024. we can see that the number of fatalities massively increased in 2023 to 2024, as 2023 have 22607 fatlities and 2024 have 24873 deaths. while it was relatively low from 2016 to 2022 as the number of fatilities was only in hundreds. so the number of fatalities did increase between different years.

## 5. Do the prices effected by the number of violence events happened in the same month?

In [78]:
months = {1: 'January', 2: 'February', 3: 'March', 4: 'April', 5: 'May', 6: 'June', 7: 'July', 8: 'August', 9: 'September', 10: 'October', 11: 'November', 12: 'December'}
df1['date'] = pd.to_datetime(df1['date']) 
df1['Month'] = df1['date'].dt.month
df1['Year'] = df1['date'].dt.year
df1['Month'] = df1.apply(lambda row: months.get(row['Month']), axis=1)
# grouping
data1 = df1.groupby(['Year', 'Month'])['price_per_kg'].mean().reset_index()
data2 = df2.groupby(['Year', 'Month'])['Number of Events'].sum().reset_index()
# mergin
merge = pd.merge(data1, data2, on=['Year','Month'], how='inner')
cor = merge2['price_per_kg'].corr(merge['Number of Events'])
print(f"the overall correlation is: {cor.round(4)}\n")

the overall correlation is: 0.6996



Yes, some prices are affected by the number of violent events that happened in the same month. though not all commodity prices are affected. such as: apples or yougurt. those commodities have consistent prices, while Fuel, Meat and wheat are more volunerable to change. The Gaza Strip by far is more affected by the number of violent events than in the west bank.